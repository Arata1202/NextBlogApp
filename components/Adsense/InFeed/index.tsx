'use client';

import { useEffect } from 'react';
import { usePathname } from 'next/navigation';

const PUBLISHER_ID = '1705865999592590';

declare global {
  interface Window {
    adsbygoogle: { [key: string]: unknown }[];
  }
}

type InFeedProps = {
  slot: string;
  format?: string;
  responsive?: string;
  style?: any;
};

const InFeed = ({ slot, format = 'fluid', responsive = 'true', style }: InFeedProps) => {
  let pathname = usePathname();
  pathname = pathname ? pathname : '';

  useEffect(() => {
    try {
      (window.adsbygoogle = window.adsbygoogle || []).push({});
    } catch (err) {
      console.error(err);
    }
  }, [pathname]);

  return (
    <div key={pathname.replace(/\//g, '-') + '-' + slot}>
      <ins
        className="adsbygoogle"
        style={{ display: 'block', width: '100%', ...style }}
        data-ad-client={`ca-pub-${PUBLISHER_ID}`}
        data-ad-layout-key="-5k+co+1n-b5+e3"
        data-ad-slot={slot}
        data-ad-format={format}
        data-full-width-responsive={responsive}
      />
    </div>
  );
};

export default InFeed;
