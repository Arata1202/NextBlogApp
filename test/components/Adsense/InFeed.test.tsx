import { render } from '@testing-library/react';
import InFeed from '@/components/Adsense/InFeed';

describe('InFeed', () => {
  test('スナップショット（InFeed)', () => {
    const { asFragment } = render(<InFeed slot="" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
