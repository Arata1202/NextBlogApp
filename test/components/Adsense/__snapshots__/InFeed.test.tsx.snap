// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InFeed スナップショット（InFeed) 1`] = `
<DocumentFragment>
  <div>
    <ins
      class="adsbygoogle"
      data-ad-client="ca-pub-1705865999592590"
      data-ad-format="fluid"
      data-ad-layout-key="-5k+co+1n-b5+e3"
      data-ad-slot="00000"
      data-full-width-responsive="true"
      style="display: block; width: 100%;"
    />
  </div>
</DocumentFragment>
`;
